olive/engine/footprint.py

```diff
@@ -134,15 +138,78 @@ def get_pareto_frontier(self):
         # restructure the pareto frontier points to instance of Footprints node for further analysis
         return Footprint(nodes=rls, objective_dict=self.objective_dict, is_marked_pareto_frontier=True)
 
-    def _plot_pareto_frontier(self, index=None):
+    def _get_metrics_name_by_indices(self, indices):
+        for _, v in self.nodes.items():
+            if v.metrics is not None and len(v.metrics.value) > 0:
+                rls = list()
+                for index in indices:
+                    if isinstance(index, str):
+                        assert index in v.metrics.value, f"the metric {index} is not in the metrics"
+                        rls.append(index)
+                    if isinstance(index, int):
+                        assert index < len(v.metrics.value), f"the index {index} is out of range"
+                        rls.append(list(v.metrics.value.keys())[index])
+                return rls
+        return None
+
+    def plot_pareto_frontier(self, index=None):
         if index is None:
             assert len(self.nodes) > 0, "you can not plot pareto frontier with empty nodes"
-            assert len(self.nodes[0].metrics.value) >= 2, "you can not plot pareto frontier with less than 2 metrics"
             index = [0, 1]
         self.mark_pareto_frontier()
+        nodes_to_be_plotted = self.get_candidates()
+
         # plot pareto frontier
-        # import plotly.graph_objects as go
-        pass
+        try:
+            import pandas as pd
+            import plotly.graph_objects as go
+
+            pd.options.mode.chained_assignment = None
+        except Exception:
+            logger.error("Please make sure you installed pandas and plotly successfully.")
```

Comment:
>error [](http://example.com/codeflow?start=19&length=5)

warning since the error could be ignored?

Reply:
Fixed.